pkg/codegen/go/gen_program.go

```diff
@@ -724,11 +724,54 @@ func (g *generator) genResource(w io.Writer, r *pcl.Resource) {
 
 }
 
+func stringsContains(values []string, value string) bool {
+	for _, item := range values {
+		if item == value {
+			return true
+		}
+	}
+	return false
+}
+
 func (g *generator) genOutputAssignment(w io.Writer, v *pcl.OutputVariable) {
 	expr, temps := g.lowerExpression(v.Value, v.Type())
+	exprArgType := ""
+
+	rewriteExpr := func() {
+		exprArgType = g.argumentTypeName(expr, expr.Type(), true)
+	}
+	switch arg := expr.(type) {
+	case *model.ScopeTraversalExpression:
+		// check if the traversal root is a resource
+		// in which case we don't need to wrap the traversal with pulumi.{Something}
+		if _, isResource := arg.Parts[0].(*pcl.Resource); !isResource {
+			rewriteExpr()
+		}
+	case *model.FunctionCallExpression:
+		nonInputtyFunctions := []string{"stack", "project", "mimeType", "join"}
```

Comment:
Can you add a comment explaining `nonImputtyFunctions`. Why these functions?

Reply:
it's commented below, the terminology is a bit weird because what I really mean is these functions do _NOT_ return `Output<T>` which seems to be `pulumi.Input` in go, at least that is what `ctx.Export` is expecting 😅